examples/website/texture-tester/app.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable react/no-unescaped-entities */
+import React, {PureComponent} from 'react';
+import {render} from 'react-dom';
+import {instrumentGLContext, Program} from '@luma.gl/core';
+import TexturesBlock from './components/textures-block';
+import {IMAGES_DATA} from './textures-data';
+
+// TEXTURE SHADERS
+
+const vs = `
+precision highp float;
+
+attribute vec2 position;
+varying vec2 uv;
+
+void main() {
+  gl_Position = vec4(position, 0.0, 1.0);
+  uv = vec2(position.x * .5, -position.y * .5) + vec2(.5, .5);
+}
+`;
+
+const fs = `
+precision highp float;
+
+uniform sampler2D tex;
+varying vec2 uv;
+
+void main() {
+  gl_FragColor = vec4(texture2D(tex, uv).rgb, 1.);
+}
+`;
+
+export default class App extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      canvas: null,
+      gl: null,
+      program: null
+    };
+
+    this.setupCanvas = this.setupCanvas.bind(this);
+    this.createAndFillBufferObject = this.createAndFillBufferObject.bind(this);
+    this.renderTexturesBlocks = this.renderTexturesBlocks.bind(this);
+    this.renderDescription = this.renderDescription.bind(this);
+  }
+
+  componentDidMount() {
+    const canvas = this.setupCanvas();
+    const gl = canvas.getContext('webgl');
+    instrumentGLContext(gl);
+    this.createAndFillBufferObject(gl);
+    const program = new Program(gl, {vs, fs});
+
+    this.setState({canvas, gl, program: program.handle});
+  }
+
+  setupCanvas() {
+    // eslint-disable-next-line no-undef
+    const canvas = document.createElement('canvas');
+    canvas.width = 256;
+    canvas.height = 256;
+    return canvas;
+  }
+
+  createAndFillBufferObject(gl) {
```

Comment:
This buffer belongs with the program. They should be created in the same file.

Reply:
It is not clear for me what you mean here.
For now we have `program creation` and `buffer creation `in the `componentDidMount` method. Could you please clarify what should I change here?